### ISSUE TITLE
Archetypes Table

### ISSUE THREAD

**Issue by ibolton336** [konveyor/tackle2-ui#1264]
Each row will represent an existing archetype, containing the following information:

- Name: Name of the archetype
- Description: Brief description of the archetype
- Tags: The list of tags that define the archetype.
- Maintainers: List of stakeholders and stakeholder groups that maintain the archetype in the organization. This is metainformation that could be useful to navigate the organization and reach out to key people when more information about an archetype is needed.
- Applications: List of applications that are associated with the archetype. Association happens automatically via a match of the tags that define with the archetype with the ones that are associated with the application.

The header of the archetypes table will include filters that include Name, Description, Tags and Maintainers as criteria. Alongside the filtering, a button will be displayed to create a new archetype.

Each row will have a menu with the following options:

Assess: Launches the assessment for a given archetype.
Review: Navigates to the review view for the archetype.
Edit: Opens a modal window to edit an existing archetype.
Delete: Deletes the archetype.

**Comment by sjd78** [comment]
After #1308, here is a list of things that need to be added to the table:

- [ ] Table action kebab and confirm modals
- [ ] Pagination (IDK if it would be client or server side)
- [ ] Table column components need to be cleaned up:
  - [ ] Description column: truncated text with tooltip for full description as needed)
  - [ ] Tags column: 
    - [ ] How to combine the criteria and archetype tags?
    - [ ] How to sort?
    - [ ] Do the tag categories matter here?
    - [ ] Reuse of the `ApplicationTagLabel` component for `Label` rendering?
  - [ ] Maintainers column:
    - [ ] Don't show the full name, generate initials
    - [ ] Sort individual stakeholders with stakeholder groups?
    - [ ] Add tooltips for each Label with the full name?
  - [ ] Application column:
    - [x] Basic "No Applications" message when no matching applications (as determined by hub response)
    - [x] Basic "Application count" message (as determined by hub response)
    - [ ] When count >0 render a link to navigate to the application inventory assessment page with filters set to show the applications for the archetype? This is not explicitly in the mockups or enhancement doc, so probably best to kick it off to a new issue.

**Comment by mturley** [comment]
I'm assuming pagination will be client side here, at least until hub folks tell us there's a scalability issue with that (I'm not sure if this endpoint supports the pagination args).

**Comment by sjd78** [comment]
With further changes detailed in JIRA tickets, this issue can be considered complete.
